src/xSTIR/pSTIR/STIR.py

```diff
@@ -470,6 +478,23 @@ def set_modality(self, mod):
         """
         return parms.set_char_par(self.handle, 'ImageData', 'modality', mod)
 
+    @property
+    def __array_interface__(self):
+        """As per https://numpy.org/doc/stable/reference/arrays.interface.html"""
+        if not self.contiguous:
+            raise ContiguousError("please make a copy first with `as_array`")
+        return {'shape': self.shape, 'typestr': '<f4', 'version': 3,
+                'data': (parms.size_t_par(self.handle, 'ImageData', 'address'), False)}
+
+    def asarray(self, xp=numpy, copy=None, **kwargs):
+        """Returns view (or fallback copy) of self"""
+        try:
+            return xp.asarray(self, copy=copy, **kwargs)
+        except ContiguousError:
+            if copy or copy is None:
+                return xp.asarray(self.as_array(), **kwargs)
+            raise
```

Comment:
note this is as per https://data-apis.org/array-api/latest/API_specification/generated/array_api.asarray.html

- `copy=None` default & behaviour
- forwarding `**kwargs`
- `ContiguousError` inherits from `ValueError`